### DISCUSSION TITLE
"Delete ALL your transactions" also removes all asset opening balance information

### DISCUSSION THREAD

**Discussion by digitlength** [firefly-iii/firefly-iii#8557] [Firefly III] [answered]
Whilst testing this software I have started over a few times, importing, playing around, then deleting and purging.

I have noticed that when using the "Delete ALL your transactions" feature, opening balance information is also removed from the asset accounts. I wasn't expecting this because I hadn't initially realised that adding opening balance information automatically creates a new transaction for that account.

So when playing around I would have to constantly re-enter opening balance information after every 'reset'.

Just wanted to check that this was expected behaviour? It took me a little while to understand what was happening because I didn't expect information which I had entered in the assets pages to also be removed.

**Comment by JC5** [comment, marked as answer]
It does yes, but it shouldn't really happen no. It happens because the opening balance is also a transaction. But as a user, you shouldn't have to care. I'll pick it up!

**Reply by JC5** [reply]
And fixed in the next release.

**Reply by digitlength** [reply]
Great stuff, thank you!

Will this fix be transparent to the user, or can we expect some slightly different behaviour? Or is Firefly just ignoring the opening balance transaction when deleting all?

**Reply by JC5** [reply]
Firefly III is just going to ignore that transaction type, yes.

**Comment by Fabeson** [comment]
@digitlength pardon my ignorance but I was also testing around and resorted to delete and recreate accounts several times to wipe faulty imports - where can I find that "Delete ALL your transactions" feature in the GUI? Thanks in advance!

**Reply by digitlength** [reply]
Menu -> Options -> Profile -> Delete and purge data

(At least for me, still on 6.3.2...)

**Reply by Fabeson** [reply]
Got it, thanks - on 6.4.15 it is called "Delete account", which 
> [...] will also delete any accounts, transactions, anything you might have saved into Firefly III. It'll be GONE. 

Not quite what I intended ;) So I will stick with the manual account deletion / re-creation for now. Cheers!

EDIT: nevermind, found the right **tab** now - thanks! :+1:
